### PR TITLE
Added models

### PR DESCRIPTION
**Pull request by InversionSpaces**
Added models to bring type safety

### CODE REVIEW

**Comment by svroonland** [comment]
Hi @InversionSpaces , thanks for the PR so far, looking good. 

Would you be willing to consider using newtypes from zio-prelude? That way the library would align with the zio ecosystem a bit better.
https://zio.github.io/zio-prelude/docs/newtypes/

**Comment by InversionSpaces** [comment]
Hi, @svroonland I'm new to CircleCI. Is it my fault that tests don't pass? They pass locally, though I'm running bare `docker run rabbitmq` and `sbt +test`. Could you help me please?

**Comment by svroonland** [comment]
Hi @InversionSpaces , not sure. It's reporting about a timeout being exceeded. Possibly the newtypes affect the speed somewhat. Could you try increasing the timeout?
https://github.com/svroonland/zio-amqp/blob/master/src/test/scala/nl/vroste/zio/amqp/AmqpClientSpec.scala#L100

**Comment by InversionSpaces** [comment]
It helped, but logs actually say that tests pass in 6 to 8 seconds, so why didnt they pass with 10 seconds timeout?

**Comment by svroonland** [comment]
Hmm I see, and it now appears to work again with 10 seconds.. Weird.

PR is looking good, let me know when it's ready to review and merge

**Comment by InversionSpaces** [comment]
Hi, @svroonland I believe this PR is ready.

File: src/main/scala/nl/vroste/zio/amqp/model/ExchangeType.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package nl.vroste.zio.amqp.model
+
+sealed trait ExchangeType extends Product with Serializable {
```
Comment (svroonland) [inline src/main/scala/nl/vroste/zio/amqp/model/ExchangeType.scala:3]: Is the `extends Product with Serializable` necessary?

**Review by svroonland — APPROVED** [review]
Great! One minor thing, besides that this is ready to merge

File: src/main/scala/nl/vroste/zio/amqp/model/ExchangeType.scala (hunk 1 above)
Comment (InversionSpaces) [reply src/main/scala/nl/vroste/zio/amqp/model/ExchangeType.scala:3]: I believe it is used to have proper type inference in situations like [this](https://scastie.scala-lang.org/6Ejz7E8EQn61DVzRgZaqdg). Here is [question on so](https://stackoverflow.com/questions/36525804/scala-case-class-extending-product-with-serializable).

**Pull request merged**